fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -927,10 +947,32 @@ ACTOR Future<Void> getResolution(CommitBatchContext* self) {
 			TenantInfo const& tenantInfo = trs[t].tenantInfo;
 			int64_t tenantId = tenantInfo.tenantId;
 			Optional<TenantNameRef> const& tenantName = tenantInfo.name;
-			// TODO(yiwu): In raw access mode, use tenant prefix to figure out tenant id for user data
 			if (tenantId != TenantInfo::INVALID_TENANT) {
 				ASSERT(tenantName.present());
 				encryptDomains[tenantId] = tenantName.get();
+			} else if (!trs[t].transaction.mutations.empty()) {
+				// For raw-access mode, parse 'tenantId' from transaction mutations by accessing first 8 bytes.
+				// For a given transaction, participant mutations can't cross tenant boundaries.
+				MutationRef const& m = trs[t].transaction.mutations[0];
+				StringRef keyRef = m.param1;
+
+				// Should this be an assert?
+				if (isSystemKey(keyRef)) {
+					encryptDomains[SYSTEM_KEYSPACE_ENCRYPT_DOMAIN_ID] =
+					    Standalone<StringRef>(StringRef(FDB_DEFAULT_ENCRYPT_DOMAIN_NAME));
+				} else {
+					int64_t tenantId = getTenantIdFromMutation(m);
+					ASSERT(tenantId != TenantInfo::INVALID_TENANT);
+					ErrorOr<Optional<TenantName>> result =
+					    getTenantName(pProxyCommitData, tenantId, true /*logOnFailure*/);
+					if (result.isError()) {
+						throw result.getError();
```

Comment:
Note that I think this means we cannot as easily erase an entire database if it is encrypted, and this is often a useful operation for test clusters. That may be ok, but it's worth considering.

Reply:
@sfc-gh-ajbeamon can you elaborate on "we cannot as easily erase an entire database if it is encrypted"? Thanks.

Reply:
If I understand correctly, this is trying to pull the tenant data for a mutation and then using it in some way. There seems to be a (maybe unvalidated) assumption here that the mutation is not a clear range that spans multiple tenants, because if it were then there would not be a single tenant that applies to the mutation.

Without the ability to do a `clearrange "" \xff`, you'll have to run a lot of separate clear ranges (possibly in separate transactions?) to do the delete.